cluster-agent/templates/deployment.yaml

```diff
@@ -1,63 +1,75 @@
-apiVersion: extensions/v1beta1
+apiVersion: apps/v1
 kind: Deployment
-metadata: 
+metadata:
   name: {{ template "cluster-agent.fullname" . }}
-spec: 
+  labels:
+    app.kubernetes.io/name: {{ include "cluster-agent.name" . }}
+    helm.sh/chart: {{ include "cluster-agent.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+spec:
   replicas: 1
-  template: 
-    metadata: 
-      annotations: 
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "cluster-agent.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+      app.kubernetes.io/managed-by: {{ .Release.Service }}
+      release: {{ .Release.Name }}
+  template:
+    metadata:
+      annotations:
         appd-agent-id: appdynamics-cluster-agent
       labels:
-        app: {{ template "cluster-agent.name" . }}
-        chart: {{ template "cluster-agent.chart" . }}
+        app.kubernetes.io/name: {{ include "cluster-agent.name" . }}
+        helm.sh/chart: {{ include "cluster-agent.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+        app.kubernetes.io/managed-by: {{ .Release.Service }}
         release: {{ .Release.Name }}
-        heritage: {{ .Release.Service }}
-    restartPolicy: Always
-    spec: 
-      containers: 
-        - env: 
+    spec:
+      restartPolicy: Always
+      containers:
+        - env:
 {{- if .Values.controller.controllerKey }}
             - name: APPDYNAMICS_AGENT_ACCOUNT_ACCESS_KEY
-              valueFrom: 
-                secretKeyRef: 
+              valueFrom:
+                secretKeyRef:
                   key: controller-key
-                  name: cluster-agent-secret
+                  name: {{ template "cluster-agent.secretName" . }}
```

Comment:
Changed hard coded name to template macro.
Secret name will also contain the release name which binds all release objects together and will allow multiple releases.